src/adaptors/mod.rs

```diff
@@ -332,7 +335,8 @@ impl<I, J> Iterator for Product<I, J>
         // Compute a * b_orig + b for both lower and upper bound
         size_hint::add(
             size_hint::mul(self.a.size_hint(), self.b_orig.size_hint()),
-            (b_min * has_cur, b_max.map(move |x| x * has_cur)))
+            (b_min * has_cur, b_max.map(move |x| x * has_cur))
+        )
```

Comment:
changes unrelated to the PR in this file unfortunatly (such should never be in the PR)